hack/vendor.sh

```diff
@@ -6,14 +6,20 @@ vendor() {
   go mod verify
 }
 
-echo "Updating dependencies for Cluster CAPI Operator"
-vendor
+vendor_in_subdir() {
+  local dir=$1
+  echo "Updating dependencies for $dir"
+  cd "$dir" && GOWORK=off vendor && cd -
+}
 
-echo "Updating dependencies for E2E tests"
-cd e2e/ && vendor && cd -
+echo "Updating dependencies for Cluster CAPI Operator"
+GOWORK=off vendor
 
-echo "Updating dependencies for hack/tools"
-cd hack/tools/ && vendor && cd -
+echo "Syncing Go workspace"
+if ! go work sync; then
+  echo "Warning: go work sync failed due to dependency conflicts. This is expected with the current vsphere provider dependency."
+  echo "The workspace structure is in place for future use, but individual module vendoring will be used for builds."
+fi
 
-echo "Updating dependencies for manifests-gen"
-cd manifests-gen/ && vendor && cd -
+vendor_in_subdir "hack/tools"
+vendor_in_subdir "manifests-gen"
```

Comment:
Shouldn't we be okay with:
* running go mod tidy for every go.mod
* running go mod verify for every go.mod
* running go work vendor

Instead of having three separate vendor directories?

Reply:
Did we come to an agreement on this one in the end @miyadav @chrischdi ?